### PR TITLE
RCPS-166:RCPS-164:RCPS-157-stud-info-mgt-app

### PR DESCRIPTION
**Pull request by Joven2002**
it includes the student email bday field and also the forgot password

### CODE REVIEW

File: src/main/java/com/student/user/app/controllers/item/ForgotPswController.java [hunk 1]
```diff
@@ -45,8 +42,35 @@ public class ForgotPswController {
...
 
 
+    private String getInvalidInputMessage() {
+        String alphanumericRegex = "[a-zA-Z0-9]+";
+
+        if ( usernameField.getText().isEmpty() || nicknameField.getText().isEmpty() || newPswField.getText().isEmpty()) {
+            return "All fields must be filled.";
+        }
+
+        if (!usernameField.getText().matches(alphanumericRegex)) {
```
Comment (zhairanicolepacheco) [inline src/main/java/com/student/user/app/controllers/item/ForgotPswController.java:52]: username field is not a scope of your ticket

File: src/main/java/com/student/user/app/controllers/item/AddStudentController.java [hunk 2]
```diff
@@ -83,6 +100,15 @@ protected void onAddStudClicked(ActionEvent event) {
             displayError("Invalid input in the following fields:", invalidFields);
             return;
         }
+        try {
```
Comment (zhairanicolepacheco) [inline src/main/java/com/student/user/app/controllers/item/AddStudentController.java:105]: this try catch was not used at all

File: src/main/java/com/student/user/app/controllers/item/AddStudentController.java [hunk 3]
```diff
@@ -171,6 +203,9 @@ private Map<String, String> getInvalidFields() {
         if (!isValidInput("Middle Name", middleName.getText())) {
             invalidFields.put("Middle Name", middleName.getText());
         }
+        if (!isValidBirthday(String.valueOf(birthday.getValue()))) {
+            invalidFields.put("Birthday", birthday.getValue().toString());
```
Comment (zhairanicolepacheco) [inline src/main/java/com/student/user/app/controllers/item/AddStudentController.java]: message should be "Student must be 17 years old and above"

**Review by zhairanicolepacheco — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/student/user/app/controllers/item/AddStudentController.java (hunk 2 above)
Comment (Joven2002) [reply src/main/java/com/student/user/app/controllers/item/AddStudentController.java:105]: this validation is blocking the user which setting the birthday year in a present year only

File: src/main/java/com/student/user/app/controllers/item/AddStudentController.java [hunk 4]
```diff
@@ -207,12 +241,27 @@ private boolean isValidInput(String fieldName, String input) {
...
     }
 
+    private boolean isValidBirthday(String birthdayString) {
+        LocalDate birthdayDate = LocalDate.parse(birthdayString);
+        if (birthdayDate == null) {
+            return false;
+        }
+
+        LocalDate currentDate = LocalDate.now();
+        LocalDate minValidDate = currentDate.minusYears(17);
```
Comment (zhairanicolepacheco) [inline src/main/java/com/student/user/app/controllers/item/AddStudentController.java:260]: What about the possibility that a student may enter college earlier due to acceleration?

**Review by zhairanicolepacheco — CHANGES_REQUESTED** [review]

**Pull request closed without merging**